File: RealmDB/app/src/main/java/org/texchtown/realmdb/MainActivity.java
package org.texchtown.realmdb;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

import io.realm.Realm;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    //var
    MainFragment mainFragment;
    FragmentManager manager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainFragment = new MainFragment();
        manager = getSupportFragmentManager();
        manager.beginTransaction().replace(R.id.container, mainFragment).commit();

    }
}

File: Alarmi/app/src/main/java/org/texchtown/alarmi/MainActivity.kt
package org.texchtown.alarmi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationBarView
import org.texchtown.alarmi.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), NavigationBarView.OnItemSelectedListener {

    private lateinit var binding: ActivityMainBinding
    private val alarmFragment = AlarmFragment()
    private val sleepingFragment = SleepingFragment()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //initialize fragment and bottom navigation
        if(savedInstanceState == null) {
            replaceFragment(alarmFragment)
            binding.bn.menu.findItem(R.id.nav_alarm).isChecked = true
        }

        binding.bn.setOnItemSelectedListener(this)
    }

    private fun replaceFragment(fragment: Fragment){
        if(fragment != null){
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.fragmentContainerView, fragment)
            transaction.commit()
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_alarm -> replaceFragment(alarmFragment)
            R.id.nav_night -> replaceFragment(sleepingFragment)
        }
        return true
    }
}